src/main/java/carparks/resources/ParkingEntryController.java

```diff
@@ -0,0 +1,88 @@
+package carparks.resources;
+
+import java.time.LocalDate;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import carparks.parking.DailySummaryDto;
+import carparks.parking.FeeType;
+import carparks.parking.ParkingEntryDto;
+import carparks.parking.ParkingService;
+
+@RestController
+class ParkingEntryController {
+	
+	private ParkingService service;
+	
+	@Autowired
+	public ParkingEntryController(ParkingService service) {
+		this.service = service;
+	}
+	
+	// ---------- Retrieve a parking entry by plate number ----------- //
+	@GetMapping(value = "/platenumber/{plateNumber}")
+	public ResponseEntity<?> getParkingEntryByPlateNumber(@PathVariable("plateNumber") String plateNumber,
+			@RequestParam(value = "currency", required = false, defaultValue = "PLN") String currency) {
+		
+		ParkingEntryDto entry = service.getEntry(plateNumber, currency);
+		
+		return new ResponseEntity<ParkingEntryDto>(entry, HttpStatus.OK);
+	}
+	
+	// ------------ Retrieve a parking entry by entry id ------------ //
+	@GetMapping(value = "/id/{entryId}")
```

Comment:
Ścieżka `/id/` trochę mało mówi o tym co jest pod nią.

Reply:
Pomysł był taki, żeby jednoznacznie rozróżnić czy dostaję entry id czy plate number, ale można np zrobić to za pomocą QueryParam, np coś takiego:
GET /entries?id={entryId}&plateNumber={plateNumber}
i rzucać wyjątkami jeśli ani id ani plate number nie jest podane albo jeśli są podane oba ale definiują różne zasoby.

Reply:
Bardziej mi chodziło o to, że bardziej wymowne by było `/entry/{entryId}`. Co myślisz o takim podejściu?

Reply:
Aaa, ok, rozważałem to od razu jak przeczytałem komentarz, ale myślałem, że to jakieś bardziej podchwytliwe pytanie. Wydaje mi się, że oba sposoby są ok, ten Twój może trochę czytelniejszy faktycznie, ale za to tez QueryParam dodatkowo weryfikuje czy id i plateNumber się zgadzają - ale też tylko w jednym przypadku, więc nie wiem czy jest sens...